shapely/geometry/polygon.py

```diff
@@ -234,24 +236,6 @@ def __new__(self, shell=None, holes=None):
             else:
                 holes = [LinearRing(ring) for ring in holes]
 
-        if not isinstance(shell, BaseGeometry):
-            if not isinstance(shell, (list, np.ndarray)):
-                # eg emtpy generator not handled well by np.asarray
-                shell = list(shell)
-            shell = np.asarray(shell)
-
-            if len(shell) == 0:
-                # empty geometry
-                # TODO better constructor + should shapely.polygons handle this?
-                return shapely.from_wkt("POLYGON EMPTY")
-
-            if not np.issubdtype(shell.dtype, np.number):
-                # conversion of coords to 2D array failed, this might be due
-                # to inconsistent coordinate dimensionality
-                raise ValueError("Inconsistent coordinate dimensionality")
-        elif not isinstance(shell, LinearRing):
-            shell = LinearRing(shell)
```

Comment:
With the small addition in `LinearRing` (to raise about the inconsistent dimensionality), this is now all handled by that constructor, so we can just call `shell = LinearRing(shell)` and simplify the code here.